src/FloatingActionButton/FloatingActionButton.js

```diff
@@ -166,6 +166,7 @@ class FloatingActionButton extends Component {
   componentWillReceiveProps(nextProps) {
     if (nextProps.disabled !== this.props.disabled) {
       this.setState({
+        hovered: nextProps.disabled ? false : this.state.hovered,
```

Comment:
I have refactored it to be more like [`FlatButton`](https://github.com/callemall/material-ui/blob/master/src/FlatButton/FlatButton.js#L121). We have two potential setState calls, but as I can see from [documentation](https://facebook.github.io/react/docs/component-specs.html#updating-componentwillreceiveprops) this will not produce any side effects

> Calling this.setState() within this function will not trigger an additional render.

On a sidenote, if using `this.state.hover` is dangerous, then we shouldn't check for it's value at all and set it to `false` by default if props.disabled is `true` (in `FlatButton` as well then). Right?